### PR TITLE
Postgres support: Detect pg gem and generate database.yml.dice

### PR DESCRIPTION
**Pull request by piao-mdsol**
## Background

`rake config:generate_all` only generates the `database.yml.dice` template file when using MySQL, it doesn't support Postgres in the same manner. We'd need to manually copy & paste the template file and update its configuration for Postgres.

With this PR, dice_bag will detect `pg` gem and generate the template file accordingly.

## Changes

- Templates for databases have been moved to `./templates/databases/{database}.yml.dice`
- `DefaultTemplateFile` now takes an `:save_as` optional argument to handle the scenario where a template filename is different than the destination filename

@mdsol/team-10

### CODE REVIEW

File: lib/dice_bag/templates/databases/postgres.yml.dice [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<%= warning.as_yaml_comment %>
+
+<% environments = Rails.env.production? ? [:production] : [:development, :test] %>
+
+<% environments.each do |env| %>
+<%= env %>:
+  adapter: <%= configured[env].database_driver || 'postgres' %>
```
Comment (piao-mdsol) [inline lib/dice_bag/templates/databases/postgres.yml.dice:7]: This file is largely same as [the mysql config](https://github.com/mdsol/dice_bag/blob/69bbaae9a4877dbb1322f16f95c338e1bc9d418a/lib/dice_bag/templates/databases/mysql.yml.dice), modified adapter, db username, encoding, and port. Let me know if any different defaults we want to bake it. 🙇

Comment (ykitamura-mdsol) [reply lib/dice_bag/templates/databases/postgres.yml.dice:7]: My eyes might be tired, but cannot find `port` 😎 :

`port: <%= configured[env].database_port || 5432 %>`

Comment (piao-mdsol) [reply lib/dice_bag/templates/databases/postgres.yml.dice:7]: Good catch..! 🙇 74ad031

**Review by ykitamura-mdsol — APPROVED** [review]
Looks good to me 👍

File: lib/dice_bag/default_template_file.rb [hunk 2]
```diff
@@ -10,12 +10,12 @@ module DiceBag
   class DefaultTemplateFile
     include DiceBagFile
 
-    def initialize(name, location=nil)
+    def initialize(name, location=nil, save_as=nil)
```
Comment (ssteeg-mdsol) [inline lib/dice_bag/default_template_file.rb:13]: How about adding spaces around the `=` sign
https://github.com/rubocop-hq/ruby-style-guide#spaces-around-equals

**Review by ssteeg-mdsol — APPROVED** [review]
Just a small comment, looks good otherwise

File: lib/dice_bag/default_template_file.rb (hunk 2 above)
Comment (piao-mdsol) [reply lib/dice_bag/default_template_file.rb:13]: Glad you mentioned it❗️ I plan to resolve the style issues in another PR, a lot of the styles in the repo don't seem to fit the coding style guide. 🙊

**Pull request merged**